Example/CollectionViewController.swift

```diff
@@ -38,23 +38,28 @@ final class CollectionViewController: UICollectionViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        self.collectionView!.register(
+        guard let collectionView = self.collectionView, let flowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout else { return }
+        collectionView.register(
             UINib(nibName: "CollectionToolCell", bundle: nil),
             forCellWithReuseIdentifier: "CollectionToolCell"
         )
 
+
+        collectionView.register(UINib(nibName: "CollectionViewHeaderView", bundle: nil), forCellWithReuseIdentifier: "CollectionViewHeaderView")
```

Comment:
this isn't a cell. it's a supplementary view.

you need `collectionView.register(, forSupplementaryViewOfKind:, withReuseIdentifier: )`

Reply:
oh wait. actually no, you shouldn't have to register explicitly. this can be removed.

Reply:
oh true